data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3ClientAuthentication.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.plugins.source.configuration.AwsAuthenticationOptions;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.sts.StsClient;
+
+public class S3ClientAuthentication {
+    private S3SourceConfig s3SourceConfig;
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+
+    public S3ClientAuthentication(final S3SourceConfig s3SourceConfig) {
+        this.s3SourceConfig = s3SourceConfig;
+        this.awsAuthenticationOptions = s3SourceConfig.getAWSAuthentication();
+    }
+
+    public S3Client createS3Client(final StsClient stsClient) {
+
+        return software.amazon.awssdk.services.s3.S3Client.builder()
```

Comment:
Can the fully qualified name be shortened from `software.amazon.awssdk.services.s3.S3Client` to `S3Client`? It looks like you already have the import.